File: Python/Functional-2/no_neg.py

# given a list of integers, return a list of the integers, omitting any that are less than 0
def no_neg(nums):
    new_list = []

    for x in nums:
        if x >= 0:
            new_list.append(x)

    return new_list

print(no_neg([1, -2]))
print(no_neg([-3, -3, 3, 3]))
print(no_neg([-1, -1, -1]))

File: JavaScript/Functional-2/noNeg.js

// given a list of integers, return a list of the integers, omitting any that are less than 0
const noNeg = (nums) => {
    const newList = nums.filter(n => n >= 0);

    return newList;
};

console.log(noNeg([1, -2]));
console.log(noNeg([-3, -3, 3, 3]));
console.log(noNeg([-1, -1, -1]));
